File: composeApp/src/desktopMain/java/ies/java/model/LocalidadId.java
package ies.java.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import org.hibernate.Hibernate;

import java.io.Serializable;
import java.util.Objects;

@Embeddable
public class LocalidadId implements Serializable {
    private static final long serialVersionUID = 8216712504036605905L;
    @Column(name = "codm", nullable = false, length = 4)
    private String codm;

    @Column(name = "provincia", nullable = false, length = 2)
    private String provincia;

    public String getCodm() {
        return codm;
    }

    public void setCodm(String codm) {
        this.codm = codm;
    }

    public String getProvincia() {
        return provincia;
    }

    public void setProvincia(String provincia) {
        this.provincia = provincia;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        LocalidadId entity = (LocalidadId) o;
        return Objects.equals(this.codm, entity.codm) &&
                Objects.equals(this.provincia, entity.provincia);
    }

    @Override
    public int hashCode() {
        return Objects.hash(codm, provincia);
    }

}

File: composeApp/src/desktopMain/java/ies/java/model/LocalidadId.kt
package ies.java.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import org.hibernate.Hibernate
import java.io.Serializable
import java.util.*

@Embeddable
open class LocalidadId : Serializable {
    @Column(name = "codm", nullable = false, length = 4)
    open var codm: String? = null

    @Column(name = "provincia", nullable = false, length = 2)
    open var provincia: String? = null
    override fun hashCode(): Int = Objects.hash(codm, provincia)
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false

        other as LocalidadId

        return codm == other.codm &&
                provincia == other.provincia
    }

    companion object {
        private const val serialVersionUID = 8896140837545460940L
    }
}